### PR TITLE
Serializer for reading Step Execution Context updated

### LINKED ISSUE #2662: Step Execution Context not provided
**Description:**
I have a SCDF task-app that internally invokes a spring batch job. This job has some steps and the steps provide execution context. This execution context is stored in the database correctly. The table BATCH_STEP_EXECUTION_CONTEXT contains my expected content. 
However if I access the job / step execution using the cloud data flow dashboard the step execution context is empty. 
In the logs of the cloud data flow server I can see the following message: 

o.s.c.d.server.batch.SimpleJobService    : Cannot load execution context for job execution: JobExecution:

After setup and debug the data flow server locally in my IDE I saw that an NullpointerException is thrown internally, which causes the message "cannot load execution context..."
I analysed that the ExecutionContextRowMapper.mapRow() method throwed that NullPointerException because the field "ExecutionContextSerializer serializer" has not been set. 

**Release versions:**
Name: spring-cloud-dataflow-server-local
Version: 2.0.0.BUILD-SNAPSHOT

Used docker compose file: 
```
version: '3'

services:
  mysql:
    image: mariadb:10.2
    environment:
      MYSQL_DATABASE: dataflow
      MYSQL_USER: root
      MYSQL_ROOT_PASSWORD: rootpw
    expose:
      - 3306
  kafka:
    image: wurstmeister/kafka:1.1.0
    expose:
      - "9092"
    environment:
      - KAFKA_ADVERTISED_PORT=9092
      - KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181
      - KAFKA_ADVERTISED_HOST_NAME=kafka
    depends_on:
      - zookeeper
  zookeeper:
    image: wurstmeister/zookeeper
    expose:
      - "2181"
  dataflow-server:
    image: springcloud/spring-cloud-dataflow-server-local:${DATAFLOW_VERSION}
    container_name: dataflow-server
    ports:
      - "9393:9393"
      - "9000-9010:9000-9010"
      - "9999"
    environment:
      - spring.cloud.dataflow.applicationProperties.stream.spring.cloud.stream.kafka.binder.brokers=kafka:9092
      - spring.cloud.dataflow.applicationProperties.stream.spring.cloud.stream.kafka.binder.zkNodes=zookeeper:2181
      - spring.cloud.skipper.client.serverUri=http://skipper-server:7577/api
      - spring.cloud.dataflow.features.schedules-enabled=false
      
      - spring.batch.initialize-schema=always
      - spring.dataflow.embedded.database.enabled=false 
      - spring.datasource.url=jdbc:mysql://mysql:3306/dataflow
      - spring.datasource.username=root
      - spring.datasource.password=rootpw
      - spring.datasource.driver-class-name=org.mariadb.jdbc.Driver
    depends_on:
      - kafka
    volumes:
      - ~/.m2:/root/.m2
  app-import:
    image: springcloud/openjdk:latest
    depends_on:
      - dataflow-server
    command: >
      /bin/sh -c "
        while ! nc -z dataflow-server 9393;
        do
          sleep 1;
        done;
        wget -qO- 'http://dataflow-server:9393/apps' --post-data='uri=http://bit.ly/Darwin-SR2-stream-applications-kafka-maven&force=true';
        echo 'Stream apps imported'
        wget -qO- 'http://dataflow-server:9393/apps' --post-data='uri=http://bit.ly/Clark-GA-task-applications-maven&force=true';
        echo 'Task apps imported'"
  skipper-server:
    image: springcloud/spring-cloud-skipper-server:${SKIPPER_VERSION}
    container_name: skipper
    ports:
    - "7577:7577"
    - "9011-9020:9011-9020"
```

**Custom apps:** 
My custom app that pupblishes step execution context can be seen here: 
https://github.com/svesch84/scdf-batch-sample

### PR DESCRIPTION
**Pull request by cppwfs**
Removed the todo
Replaced the XStream serializer with Jackson2ExecutionContextStringSerializer

resolves #2662

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobServiceFactoryBean.java [hunk 1]
```diff
@@ -215,13 +216,10 @@ public void afterPropertiesSet() throws Exception {
...
 		}
 
-		// TODO: BOOT2 this may cause issues as it just removed for now
-		//if (serializer == null) {
-		//	XStreamExecutionContextStringSerializer defaultSerializer = new XStreamExecutionContextStringSerializer();
-		//	defaultSerializer.afterPropertiesSet();
-		//
-		//	serializer = defaultSerializer;
-		//}
+		if (serializer == null) {
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobServiceFactoryBean.java]: Can we have a test that verifies this, please?

Comment (cppwfs) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobServiceFactoryBean.java]: Added

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Rebased and merged as 83a796a3

**Pull request closed without merging**